internal/core/axis.cpp

```diff
@@ -49,3 +50,58 @@ bool Axis::inrange(float coordinate) const noexcept(true) {
 float Axis::to_sample_position(float coordinate) noexcept(false) {
     return this->m_axis_descriptor.CoordinateToSamplePosition(coordinate);
 }
+
+void Axis::validate_compatible(Axis const& other) noexcept(false) {
```

Comment:
Name is weird. What does "compatible" mean?
You do not explain in the docs or in the commit description (commit is named "validate axis" btw - you do not add any "validate axis" here)
Need to think if that is equality or not. Should this be just operator == that would return true/false and something else will throw? Should it be validate/assert equal? Should that be an external function instead, if "compatible" makes sense only in certain context?